### DISCUSSION TITLE
How to define channel specific configuration in nats-jetstream

### DISCUSSION THREAD

**Discussion by anil1996** [nats-io/nats-server#3415] [Q&A] [answered]
Hello,

We are planning to move to nats-jetstream from nats-streaming.
Is there any option available to define channel specific configuration in jetstream, like how it is available in nats-streaming?
For example: 
stan.conf content

# Maximum Payload allowed 
max_payload: 4MB
# stan message bus configuration file
store_limits: {
    # Override some global limits
    max_msgs: 200 
    # Global limit for all topics
    max_bytes: 128MB 
    max_age: "5m"

    # Per channel configuration.
    # Can be channels, channels_limits, per_channel, per_channel_limits or ChannelsLimits
    channels: {
        "test1": {
            # Possible options are the same than in the store_limits section, except
            # for max_channels. Not all limits need to be specified.
            max_msgs: 0
        }
    }
}

how can I achieve same configuration in jetstream???

**Comment by ripienaar** [comment]
No unfortunately not, you have to use the API, CLI, Terraform or GitHub Actions

**Reply by anil1996** [reply]
Hi @ripienaar Thanks for your response. 

As per my understanding it's possible to configure max_msgs, max_bytes, and max_age. It also seems possible to set max-msgs-per-subject, but this configuration max-msgs-per-subject sets it uniformly for all subjects in a stream. Can I have customized values for different subjects ?
for eg: stream A contains test1 and test2 subjects.
is it possible to set max-msgs-per-subject 2 for test1 and 3 for test2??

**Reply by ripienaar** [reply, marked as answer]
No, as you say its per stream

**Reply by anil1996** [reply]
Ok Thank you.
So I  have to create different different streams.
One more clarification needed.
I have chosen storage to be memory , I have 100 streams 1 replica and I am restricting each stream with max bytes as 128MB. When I start server with jetstream enabled I can see max memory as 8.77GB but my configuration file looks like this

jetstream {
  max_file: 0
}

since max memory is 8.77GB and I have 100 streams with max bytes as 128MB, so combined memory is ~12GB.
While creating streams using go clients I am getting insufficient resource error.

So does nats pre allocates memory for all these streams or its just a check to ensure that I have enough memory combined to create all these streams???
I checked RAM usage for nats-server and its around ~34.4 MiB.
